sympy/series/tests/test_formal.py

```diff
@@ -377,38 +377,43 @@ def test_fps__logarithmic_singularity_fail():
     assert fps(f, x) == log(2) - log(x) - x**2/4 - 3*x**4/64 + O(x**6)
 
 
-@XFAIL
-def test_fps__symbolic():
+def test_fps_symbolic():
     f = x**n*sin(x**2)
     assert fps(f, x).truncate(8) == x**2*x**n - x**6*x**n/6 + O(x**(n + 8), x)
 
-    f = x**(n - 2)*cos(x)
-    assert fps(f, x).truncate() == \
-        (x**n*(-S(1)/2 + x**(-2)) + x**2*x**n/24 - x**4*x**n/720 +
-         O(x**(n + 6), x))
-
     f = x**n*log(1 + x)
     fp = fps(f, x)
     k = fp.ak.variables[0]
     assert fp.infinite == \
         Sum((-(-1)**(-k)*x**k*x**n)/k, (k, 1, oo))
 
-    f = x**(n - 2)*sin(x) + x**n*exp(x)
-    assert fps(f, x).truncate() == \
-        (x**n*(1 + 1/x) + 5*x*x**n/6 + x**2*x**n/2 + 7*x**3*x**n/40 +
-         x**4*x**n/24 + 41*x**5*x**n/5040 + O(x**(n + 6), x))
-
     f = (x - 2)**n*log(1 + x)
     assert fps(f, x, 2).truncate() == \
-        ((x - 2)**n*log(3) - (x - 2)**2*(x - 2)**n/18 +
+        ((x - 2)**n*log(3) + (x - 2)*(x - 2)**n/3 - (x - 2)**2*(x - 2)**n/18 +
          (x - 2)**3*(x - 2)**n/81 - (x - 2)**4*(x - 2)**n/324 +
-         (x - 2)**5*(x - 2)**n/1215 + (x/3 - S(2)/3)*(x - 2)**n +
-         O((x - 2)**(n + 6), (x, 2)))
+         (x - 2)**5*(x - 2)**n/1215 + O((x - 2)**(n + 6), (x, 2)))
+
+    f = x**(n - 2)*cos(x)
+    assert fps(f, x).truncate() == \
+        (x**n*x**(-2) - x**n/2 + x**2*x**n/24 - x**4*x**n/720 + O(x**(n + 6), x))
+
+    f = x**(n - 2)*sin(x) + x**n*exp(x)
+    assert fps(f, x).truncate() == \
+        (x**n/x + x**n + 5*x*x**n/6 + x**2*x**n/2 + 7*x**3*x**n/40 +
+         x**4*x**n/24 + 41*x**5*x**n/5040 + O(x**(n + 6), x))
```

Comment:
Is it possible to display results like instead of ``x**4*x**n/24`` => ``x**(n+4)/24``. IMHO it might just be a simple`` powsimp`` call at the right place.